edgraph/server.go

```diff
@@ -823,6 +832,17 @@ func findMutationVars(qc *queryContext) []string {
 		varsList = append(varsList, v)
 	}
 
+	// We use certain variables to check uniqueness. To prevent errors related to unused
+	// variables, we explicitly add them to the 'varsList' since they are not used in the mutation.
+	for _, uniqueQueryVar := range qc.uniqueVars {
```

Comment:
Why aren't we update the mutation and letting dgraph doing it for us? We won't need this code then.

Reply:
How will we check the data in badger before the mutation if we don't create these dummy variables?